src/foam.js

```diff
@@ -15,39 +15,27 @@
  * limitations under the License.
  */
 
-// TODO:
-//  break into foam.js for web and foam_node.js for node.js
-//  change flag defaults for each
-//     default: web, debug, js
-//     node: node, java, swift, debug, js
-//  ???: is 'js' flag used anywhere, if so, where?
-//  replace use of global and window with new globalThis
-//  replace all use of 'global' in foam with globalThis
-//  remove support for workers
-//  case-specific setting of flags (HTTP parameters or command-line)
 
 (function() {
-
-  var isWorker = typeof importScripts !== 'undefined';
-  var isServer = ( ! isWorker ) && typeof window === 'undefined';
-
-  // Imports used by the loadServer() loader
-  if ( isServer && typeof global !== 'undefined' ) {
-    global.imports = {}; global.imports.path = require('path');
-  }
-
   var flags    = this.FOAM_FLAGS = this.FOAM_FLAGS || {};
-  flags.web    = ! isServer;
-  flags.node   = isServer;
-  flags.loader = ! isServer;
+  flags.node   = false;
+  flags.web    = true;
   if ( ! flags.hasOwnProperty('java')  ) flags.java  = false;
   if ( ! flags.hasOwnProperty('swift') ) flags.swift = false;
   if ( ! flags.hasOwnProperty('debug') ) flags.debug = true;
   if ( ! flags.hasOwnProperty('js')    ) flags.js    = true;
 
+  // set flags by url parameters
+  var urlParams = new URLSearchParams(globalThis.location.search);
+  if ( urlParams.get('node')  ) flags.node = urlParams.get('node') === 'true';
+  if ( urlParams.get('web')   ) flags.web = urlParams.get('web') !== 'false';
+  if ( urlParams.get('java')  ) flags.java = urlParams.get('java') === 'true';
+  if ( urlParams.get('swift') ) flags.swift = urlParams.get('swift') === 'true';
+  if ( urlParams.get('debug') ) flags.debug = urlParams.get('debug') !== 'false';
+  if ( urlParams.get('js')    ) flags.js = urlParams.get('js') !== 'false';
+
   function createLoadBrowser() {
```

Comment:
Rename to createLoader(), no need to call createLoadBrowser() since there is only one version now.